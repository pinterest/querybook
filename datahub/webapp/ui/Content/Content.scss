.Content {
    li + li {
        margin-top: 0.25em;
    }

    p:not(:last-child),
    dl:not(:last-child),
    ol:not(:last-child),
    ul:not(:last-child),
    blockquote:not(:last-child),
    pre:not(:last-child),
    table:not(:last-child) {
        margin-bottom: 1em;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        font-weight: 600;
        line-height: 1.125;
    }

    h1 {
        font-size: 2em;
        margin-bottom: 0.5em;
    }

    h1:not(:first-child) {
        margin-top: 1em;
    }

    h2 {
        font-size: 1.75em;
        margin-bottom: 0.5714em;
    }

    h2:not(:first-child) {
        margin-top: 1.1428em;
    }

    h3 {
        font-size: 1.5em;
        margin-bottom: 0.6666em;
    }

    h3:not(:first-child) {
        margin-top: 1.3333em;
    }

    h4 {
        font-size: 1.25em;
        margin-bottom: 0.8em;
    }

    h5 {
        font-size: 1.125em;
        margin-bottom: 0.8888em;
    }

    h6 {
        font-size: 1em;
        margin-bottom: 1em;
    }

    blockquote {
        background-color: var(--light-bg-color);
        border-left: var(--border-color);
        padding: 1.25em 1.5em;
    }

    ol {
        list-style-position: outside;
        margin-left: 2em;
        margin-top: 1em;
    }

    ul {
        list-style: disc outside;
        margin-left: 2em;
        margin-top: 1em;
    }

    pre {
        -webkit-overflow-scrolling: touch;
        overflow-x: auto;
        padding: 1.25em 1.5em;
        white-space: pre;
        word-wrap: normal;
    }
}
