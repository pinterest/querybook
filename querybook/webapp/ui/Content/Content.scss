.Content {
    li + li {
        margin-top: 4px;
    }

    p:not(:last-child),
    dl:not(:last-child),
    ol:not(:last-child),
    ul:not(:last-child),
    blockquote:not(:last-child),
    pre:not(:last-child),
    table:not(:last-child) {
        margin-bottom: 12px;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        color: var(--dark-text-color);
        font-weight: 600;
        line-height: 1.125;
    }

    h1 {
        font-size: var(--xlarge-text-size);
        margin-bottom: 8px;
    }
    h1:not(:first-child) {
        margin-top: 16px;
    }

    h2 {
        font-size: 1.75em;
        margin-bottom: 9px;
    }
    h2:not(:first-child) {
        margin-top: 18px;
    }

    h3 {
        font-size: var(--large-text-size);
        margin-bottom: 10px;
    }
    h3:not(:first-child) {
        margin-top: 20px;
    }

    h4 {
        font-size: var(--med-text-size);
        margin-bottom: 12px;
    }

    h5 {
        font-size: 1.125em;
        margin-bottom: 14px;
    }

    h6 {
        font-size: var(--text-size);
        margin-bottom: 16px;
    }

    blockquote {
        background-color: var(--light-bg-color);
        border-left: var(--border-color);
        padding: 24px;
    }

    pre {
        font-size: inherit;
    }

    ol {
        list-style-position: outside;
        margin-left: 32px;
        margin-top: 16px;
    }

    ul {
        list-style: disc outside;
        margin-left: 32px;
        margin-top: 16px;
    }
    ul ul {
        list-style-type: circle;
        margin-top: 8px;
    }
    ul ul ul {
        list-style-type: square;
    }

    img {
        display: block;
        width: 720px;
        margin: 12px 4px;
        border: var(--border);
        padding: 4px;
    }

    table {
        border-collapse: collapse;
        th {
            color: var(--title-color);
            border: 1px solid var(--focus-border-color);
            padding: 4px;
        }
        td {
            color: var(--text-color);
            border: 1px solid var(--focus-border-color);
            padding: 4px;
        }
    }
}
