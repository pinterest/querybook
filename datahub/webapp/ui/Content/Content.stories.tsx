import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Content } from './Content';

export default {
    title: 'Stateless/Content',
    decorators: [centered],
};
export const _Content = () => (
    <Content>
        <h1>Content styles HTML</h1>
        <p>paragraph</p>
        <ul>
            <li>
                <u>list item 1</u>
            </li>
            <li>
                <i>list item 2</i>
            </li>
            <li>
                <b>list item 3</b>
            </li>
        </ul>
        <code>code</code>
    </Content>
);
