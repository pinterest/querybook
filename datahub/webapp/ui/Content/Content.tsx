import styled from 'styled-components';
import './Content.scss';

export const Content = styled.div.attrs({
    className: 'Content',
})``;
