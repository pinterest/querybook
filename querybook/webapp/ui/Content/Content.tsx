import styled from 'styled-components';

import './Content.scss';

export const Content = styled.div.attrs({
    className: 'Content',
})`
    h1,
    h2 {
        font-family: var(--font-accent);
        letter-spacing: var(--letter-spacing);
    }
`;
