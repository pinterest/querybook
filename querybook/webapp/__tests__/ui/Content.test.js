import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { Content } from '../../ui/Content/Content';

it('renders without crashing', () => {
    shallow(<Content dangerouslySetInnerHTML={{ __html: '<p>test</p>' }} />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(
            <Content dangerouslySetInnerHTML={{ __html: '<p>test</p>' }} />
        );
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(
            <Content dangerouslySetInnerHTML={{ __html: '<p>test</p>' }} />
        );
        expect(output).toMatchSnapshot();
    });
});
