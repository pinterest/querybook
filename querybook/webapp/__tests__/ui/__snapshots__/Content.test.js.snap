// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="Content-sc-13ypr6e-0 Content"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test</p>",
    }
  }
/>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="Content-sc-13ypr6e-0 Content"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test</p>",
    }
  }
/>
`;
